tests/unit/deploy/test_packager.py

```diff
@@ -46,6 +46,26 @@ def pip_runner():
 
 
 @pytest.fixture
+def custom_env_osutils():
+    class CustomEnv(object):
+        def __init__(self, env):
+            self._env = env
+
+        def environ(self):
+            return self._env
+    return CustomEnv
+
+
+@pytest.fixture
+def pip_runner_custom_env(custom_env_osutils):
+    def create_pip_runner(env):
```

Comment:
So idea. What if instead of making a bunch of different pip runners, we just made a `pip_runner_factory` fixture? That way if we add anything more to the `PipRunner()` class, we do not have to add a new fixture for each of these.

Reply:
Yes I mainly didn't want to change the interface used in every single test.